File: src/centrosalud/AtencionMedica.java
package centrosalud;

public class AtencionMedica {
    private String idAtencion;
    private String diagnostico;
    private String tratamiento;
    private String observaciones;
    private Receta receta;

    public AtencionMedica(String idAtencion, String diagnostico,
                          String tratamiento, String observaciones) {
        this.idAtencion = idAtencion;
        this.diagnostico = diagnostico;
        this.tratamiento = tratamiento;
        this.observaciones = observaciones;
        this.receta = new Receta(); // toda atención nace con su propia receta, vacía
    }

    public String getIdAtencion() { return idAtencion; }
    public String getDiagnostico() { return diagnostico; }
    public String getTratamiento() { return tratamiento; }
    public String getObservaciones() { return observaciones; }
    public Receta getReceta() { return receta; }

    // Delega en Receta: la atención no maneja el stock ella misma,
    // se lo pasa a SU receta, que es quien realmente controla eso.
    public void agregarMedicamento(Medicamento medicamento) {
        receta.agregarMedicamento(medicamento);
    }

    public void mostrarAtencion() {
        System.out.println("ID Atención: " + idAtencion);
        System.out.println("Diagnóstico: " + diagnostico);
        System.out.println("Tratamiento: " + tratamiento);
        System.out.println("Observaciones: " + observaciones);
        receta.mostrarMedicamentos();
    }
}

File: modelo/atencion_medica.py
class AtencionMedica:
    def __init__(self, id_atencion: str, diagnostico: str,
                 tratamiento: str, observaciones: str):
        if not id_atencion or not id_atencion.strip():
            raise ValueError("El id de la atención es obligatorio")
        if not diagnostico or not diagnostico.strip():
            raise ValueError("El diagnóstico es obligatorio")
        self._id_atencion = id_atencion.strip()
        self._diagnostico = diagnostico.strip()
        self._tratamiento = tratamiento
        self._observaciones = observaciones

    @property
    def id_atencion(self) -> str:
        return self._id_atencion

    @property
    def diagnostico(self) -> str:
        return self._diagnostico

    @property
    def tratamiento(self) -> str:
        return self._tratamiento

    @property
    def observaciones(self) -> str:
        return self._observaciones

    def mostrar_atencion(self) -> None:
        print(f"ID Atención: {self._id_atencion}")
        print(f"Diagnóstico: {self._diagnostico}")
        print(f"Tratamiento: {self._tratamiento}")
        print(f"Observaciones: {self._observaciones}")
